lib/tokenlist.cpp

```diff
@@ -2225,24 +2225,28 @@ void TokenList::simplifyStdType()
 bool TokenList::isKeyword(const std::string &str) const
 {
     if (isCPP()) {
-        // TODO: integrate into keywords?
-        // types and literals are not handled as keywords
-        static const std::unordered_set<std::string> cpp_types = {"bool", "false", "true"};
-        if (cpp_types.find(str) != cpp_types.end())
-            return false;
-
         const auto &cpp_keywords = Keywords::getAll(mSettings.standards.cpp);
-        return cpp_keywords.find(str) != cpp_keywords.end();
+        const bool b = cpp_keywords.find(str) != cpp_keywords.end();
+        if (b) {
+            // TODO: integrate into keywords?
+            // types and literals are not handled as keywords
+            static const std::unordered_set<std::string> cpp_types = {"bool", "false", "true"};
```

Comment:
I am curious why this does not include the types we filter out with C code. But out of the scope of this change.